### DISCUSSION TITLE
Question: When does Payload plan to bump its graphql version?

### DISCUSSION THREAD

**Discussion by andrewjroberts** [payloadcms/payload#1202] [Roadmap]
Hi all,

Short question — title says it all — is there a timeline around when Payload expects to upgrade to graphql@16? 

My team tried to migrate to Apollo Federation, but a lot of the Apollo tooling uses graphql@16 and Payload is pinned to v15 (along with TypeGraphQL, for that matter) so we're holding off.

Thank you!

**Comment by jmikrut** [comment]
Hey @andwrobs — good question!

I haven't looked into what's needed to migrate to 16 but this is something that I will tag as `planned` and we will make it happen, hopefully within the next few months here. We'd also accept a PR to do this (might be easy, might just need trivial code changes).

I'll report back here once I know more but in the meantime, if you need this fast, maybe you could take a crack at updating and giving it a shot?

**Reply by andrewjroberts** [reply]
We'll dig in on our end too, see who replies first. Thanks @jmikrut !

**Reply by andrewjroberts** [reply]
From our end, we were able to work around by creating our Federation supergraph through Apollo Studio — i.e. not in code, so the GraphQL dependency mismatch didn't matter.

**Comment by andrewjroberts** [comment]
for anyone previously watching, there's an open PR for this

https://github.com/payloadcms/payload/pull/1657
